### PR TITLE
W1 - D4 - Event - Listener

### PR DESCRIPTION
**Pull request by at-dinhtruong**

### CODE REVIEW

File: app/src/at-dinhtruong/java/asiantech/internship/summer/EventAndListenerActivity.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    private RadioButton mRbMale;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_event_and_listener);
+        init();
+    }
+
+    private void init() {
```
Comment (at-vinhhuynh) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/EventAndListenerActivity.java]: method name to `initViewsAndListener`

File: app/src/at-dinhtruong/java/asiantech/internship/summer/EventAndListenerActivity.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        @Override
+        public void afterTextChanged(Editable s) {
+            checkAll();
+        }
+    };
+
+    private boolean isValidUsername(String mUsername) {
+        Pattern patternUsername;
+        Matcher matcherUsername;
+        final String USERNAME_PATTERN = "^.{6,18}$";
```
Comment (at-vinhhuynh) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/EventAndListenerActivity.java]: define as a global variable

File: app/src/at-dinhtruong/java/asiantech/internship/summer/EventAndListenerActivity.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        patternUsername = Pattern.compile(USERNAME_PATTERN);
+        matcherUsername = patternUsername.matcher(mUsername);
+        return matcherUsername.matches();
+
+    }
+
+    private boolean isValidPassword(String mPassword) {
+        Pattern patternPassword;
+        Matcher matchePasswordr;
+        final String PASSWORD_PATTERN = "^(?=.*[0-9])(?=.*[a-zA-Z]).{6,18}$";
```
Comment (at-vinhhuynh) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/EventAndListenerActivity.java]: define as a global variable

File: app/src/at-dinhtruong/java/asiantech/internship/summer/EventAndListenerActivity.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        patternPassword = Pattern.compile(PASSWORD_PATTERN);
+        matchePasswordr = patternPassword.matcher(mPassword);
+        return matchePasswordr.matches();
+
+    }
+
+    private boolean isValidEmail(String mEmail) {
+        Pattern patternEmail;
+        Matcher matcherEmail;
+        String emailPattern = "^([a-zA-Z0-9_\\-.]+)@((\\[[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.)|(([a-zA-Z0-9\\-]+\\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(]?)$";
```
Comment (at-vinhhuynh) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/EventAndListenerActivity.java]: define as a global variable

File: app/src/at-dinhtruong/java/asiantech/internship/summer/EventAndListenerActivity.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        patternEmail = Pattern.compile(emailPattern);
+        matcherEmail = patternEmail.matcher(mEmail);
+        return matcherEmail.matches();
+    }
+
+    private void checkAll() {
+        String mUsernameInput = mEdtUsername.getText().toString().trim();
```
Comment (at-vinhhuynh) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/EventAndListenerActivity.java]: please change variable name flow convention

File: app/src/at-dinhtruong/java/asiantech/internship/summer/EventAndListenerActivity.java [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        String mPasswordInput = mEdtPassword.getText().toString().trim();
+        String mEmailInput = mEdtEmail.getText().toString().trim();
+        if (isValidUsername(mUsernameInput) && isValidPassword(mPasswordInput) && isValidEmail(mEmailInput) && (mRbMale.isChecked() || mRbFemale.isChecked())) {
+            mImgApply.setVisibility(View.VISIBLE);
+        } else if (!isValidUsername(mUsernameInput)) {
+            mImgApply.setVisibility(View.GONE);
+            mEdtUsername.setError("User Format Wrong!");
```
Comment (at-vinhhuynh) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/EventAndListenerActivity.java]: define all string (ex: `"User Format Wrong!"`) on string resource

File: app/src/at-dinhtruong/java/asiantech/internship/summer/EventAndListenerActivity.java [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+        }
+
+        @Override
+        public void onTextChanged(CharSequence s, int start, int before, int count) {
+
+        }
+
+        @Override
+        public void afterTextChanged(Editable s) {
+            checkAll();
```
Comment (at-vinhhuynh) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/EventAndListenerActivity.java]: change method name more clearly

File: app/src/at-dinhtruong/res/values/strings.xml [hunk 8]
```diff
@@ -1,3 +1,11 @@
 <resources>
     <string name="app_name">Intern18Summer</string>
+    <string name="user_name">Password</string>
+    <string name="sign_up_activity">SignUpActivity</string>
+    <string name="pass_word">Password</string>
+    <string name="email">Email</string>
+    <string name="male">Male</string>
+    <string name="female">Female</string>
+    <string name="email_subscriptions">Email subscriptions</string>
```
Comment (at-vinhhuynh) [inline app/src/at-dinhtruong/res/values/strings.xml]: `email_subscriptions` wrong convention

Comment (at-vinhhuynh) [reply app/src/at-dinhtruong/res/values/strings.xml]: also check other resource name

**Comment by at-dinhtruong** [comment]
I fixed comments. Please review it!

File: app/src/at-dinhtruong/java/asiantech/internship/summer/EventAndListenerActivity.java [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+        @Override
+        public void afterTextChanged(Editable s) {
+            checkUserPassEmail();
+        }
+    };
+
+    private boolean isValidUsername(String mUsername) {
+        Pattern patternUsername;
+        Matcher matcherUsername;
+        patternUsername = Pattern.compile(USERNAME_PATTERN);
```
Comment (at-vinhhuynh) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/EventAndListenerActivity.java]:
```
Pattern  patternUsername = Pattern.compile(USERNAME_PATTERN); 
Matcher  matcherUsername = patternUsername.matcher(mUsername);
```
Apply for other method

**Pull request merged**